src/bun.js/VirtualMachine.zig

```diff
@@ -1555,11 +1555,11 @@ fn _resolve(
         return;
     } else if (strings.hasPrefixComptime(specifier, js_ast.Macro.namespaceWithColon)) {
         ret.result = null;
-        ret.path = specifier;
+        ret.path = try bun.default_allocator.dupe(u8, specifier);
```

Comment:
these are allocated now because the caller of this function frees `specifier` at the end of the function if it's allocated. non-ascii triggers the bug because we need to allocate for the convertion to utf8